contracts/issuance/IssuanceEth.sol

```diff
@@ -139,12 +140,14 @@ contract IssuanceEth is Ownable, StateMachine, ReentrancyGuard {
     /**
      * @dev Function to transfer all collected tokens to the wallet of the owner
      */
-    function transferFunds(address payable _wallet) public onlyOwner {
+    function withdraw(address payable _wallet) public onlyOwner nonReentrant {
         require(
             currentState == "LIVE",
             "Cannot transfer funds now."
         );
-        _wallet.transfer(amountRaised);
+        uint256 amount = amountRaised - amountWithdrawn;
+        amountWithdrawn = amount;
```

Comment:
Shouldn't we have a `require(amount > 0, "Nothing to withdraw.");` here? If it goes through `_wallet.transfer(amount)`, it will still use some ether from the issuance (possibly some dividends) as gas, even if we transfer 0 ether.

Reply:
I'm not sure if that is how it works, why would the contract pay for gas? I would assume msg.sender pays for gas, including the 2300 to execute `transfer`.

Reply:
@obernardovieira?

Reply:
Yes, you are right.